website_animate/static/src/js/o_animate.frontend.js

```diff
@@ -1,33 +1,48 @@
 odoo.define('website_animate.o_animate_frontend', function (require) {
 'use strict';
 
-var publicWidget = require('web.public.widget');
+const publicWidget = require('web.public.widget');
 
-var WebsiteAnimate = {
-    win   : {},
-    items : {},
+publicWidget.registry.WebsiteAnimate = publicWidget.Widget.extend({
+    selector: '#wrapwrap',
+    disabledInEditableMode: false,
+    win: {},
+    items: {},
 
-    offsetRatio : 0.3,  // Dynamic offset ratio: 0.3 = (element's height/3)
-    offsetMin   : 10,   // Minimum offset for small elements (in pixels)
+    offsetRatio: 0.3, // Dynamic offset ratio: 0.3 = (element's height/3)
+    offsetMin: 10, // Minimum offset for small elements (in pixels)
 
-    // Retrieve animable elements and attach handlers.
+    /**
+     * @override
+     */
     start: function () {
-        var self   = this;
+        // By default, elements are hidden by the css of o_animate.
+        // render elements  + // We will trigger the animation then pause it in state 0.
+        const self = this;
         self.items = $(".o_animate");
         self.items.each(function () {
-            var $el = $(this);
+            const $el = $(this);
             // Set all monitored elements to initial state
             self.reset_animation($el);
         });
         setTimeout(function () {
             self.attach_handlers();
         });
+        // Then we render all the elements, the ones which are invisible
+        // in state 0 (like fade_in for example) will stay invisible.
+        $(".o_animate").css("visibility", "visible");
```

Comment:
Is there any reason for the widget to be bound to `#wrapwrap` instead of each individual animation? That would avoid searching for them all here, looping through them, etc.

Reply:
I think I have to leave it like that because otherwise animations restart each time an options is changed in the snippet..
Because of the the "start" method of the public widget on ".o_animate" .